ext/opentelemetry-ext-http-requests/tests/test_requests_integration.py

```diff
@@ -114,20 +69,84 @@ def test_invalid_url(self):
             exception_type = ValueError
 
         with self.assertRaises(exception_type):
-            requests.post(url=url)
-        call_args = (
-            self.tracer.start_as_current_span.call_args  # pylint:disable=no-member
-        )
-        self.assertTrue(
-            call_args[0][0].startswith("<Unparsable URL"),
-            msg=self.tracer.start_as_current_span.call_args,  # pylint:disable=no-member
-        )
-        self.span_context_manager.__enter__.assert_called_with()
-        exitspan = self.span_context_manager.__exit__
-        self.assertEqual(1, len(exitspan.call_args_list))
-        self.assertIs(exception_type, exitspan.call_args[0][0])
-        self.assertIsInstance(exitspan.call_args[0][1], exception_type)
+            requests.post(url)
+
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+        span = span_list[0]
+
+        self.assertTrue(span.name.startswith("<Unparsable URL"))
         self.assertEqual(
-            self.span_attrs,
+            span.attributes,
             {"component": "http", "http.method": "POST", "http.url": url},
         )
+
+    def test_disable(self):
+        opentelemetry.ext.http_requests.disable()
+        result = requests.get(self.URL)
+        self.assertEqual(result.text, "Hello!")
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 0)
+
+    def test_disable_session(self):
+        session1 = requests.Session()
+        opentelemetry.ext.http_requests.disable_session(session1)
+
+        result = session1.get(self.URL)
+        self.assertEqual(result.text, "Hello!")
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 0)
+
+        # Test that other sessions as well as global requests is still
+        # instrumented
+        session2 = requests.Session()
+        result = session2.get(self.URL)
+        self.assertEqual(result.text, "Hello!")
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+
+        self.memory_exporter.clear()
+
+        result = requests.get(self.URL)
+        self.assertEqual(result.text, "Hello!")
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+
+    def test_suppress_instrumentation(self):
```

Comment:
unrelated to this PR, but you reminded me that we should handle instrumentation suppression on a different layer: https://github.com/open-telemetry/opentelemetry-python/issues/581. Then we can remove this test.